kite-data/kite-data-core/src/main/java/org/kitesdk/data/DatasetDescriptor.java

```diff
@@ -486,7 +487,7 @@ public Builder schemaLiteral(String s) {
      * @since 0.2.0
      */
     public <T> Builder schema(Class<T> type) {
-      this.schema = ReflectData.get().getSchema(type);
+      this.schema = DataModelUtil.getTypeSchema(type);
```

Comment:
This breaks tests because it uses the `AllowNull` schema instead of assuming fields are generally non-null and annotated with `@Nullable` if necessary. This doesn't affect the behavior with any data already written, so it is a backward-compatible change. It is also a valid evolution because fields become more permissible. I think this would require CDK-986 (#353) but otherwise is pretty safe.